src/js/components/deployments/deployment-wizard/phasesettings.js

```diff
@@ -87,56 +90,53 @@ export default class PhaseSettings extends React.Component {
-                  onChange={event => self.updateBatchSize(event.target.value, index)}
-                  endAdornment={<InputAdornment position="end">%</InputAdornment>}
-                  inputProps={{
-                    step: 1,
-                    min: 1,
-                    max: max,
-                    type: 'number'
-                  }}
-                />
-              ) : (
-                phase.batch_size || remainder
-              )}
-              <span className="info" style={{ marginLeft: '5px' }}>{`(${deviceCount} ${pluralize('device', deviceCount)})`}</span>
-            </TableCell>
-            <TableCell>
-              <Time value={startTime} format="YYYY-MM-DD HH:mm" />
-            </TableCell>
-            <TableCell>
-              {phase.delay && index !== props.phases.length - 1 ? (
+    // disable 'add phase' button if last phase/remainder has only 1 device left
+    const disableAdd = ((remainder/100)*props.numberDevices ) <= 1;
+    const phases = props.phases ? props.phases.map((phase, index) => {
+      let max = index > 0 ? 100-(props.phases[index-1].batch_size) : 100;
+      const deviceCount = (index === props.phases.length-1) 
+        ? Math.ceil((props.numberDevices / 100) * (phase.batch_size || remainder))
+        : Math.floor((props.numberDevices / 100) * phase.batch_size);
+
+      const startTime = !(index && phase.start_ts) ? new Date().toISOString() : phase.start_ts;
+      return (
+        <TableRow key={index}>
+          <TableCell component="th" scope="row">
+            <Chip size="small" label={`Phase ${index+1}`} />
+          </TableCell>
+          <TableCell>{phase.batch_size && (phase.batch_size<100) ? 
+            <Input
+              value={phase.batch_size}
+              margin="dense"
+              onChange={event => self.updateBatchSize(event.target.value, index)}
+              endAdornment={<InputAdornment className={(deviceCount<1) ? 'warning' : ''} position="end">%</InputAdornment>}
+              disabled={(self.props.disabled && deviceCount>=1) ? true : false}
+              inputProps={{
+                step: 1,
+                min: 1,
+                max: max,
+                type: 'number',
+              }}
+            />
+            : phase.batch_size || remainder}
+          <span className={(deviceCount<1) ? 'warning info' : 'info'} style={{marginLeft:'5px'}}>{`(${deviceCount} ${pluralize('device', deviceCount)})`}</span>
+
+          {(deviceCount<1) ? <div className="warning">Phases must have at least 1 device</div> : null }
+          </TableCell>
+          <TableCell><Time value={startTime} format="YYYY-MM-DD HH:mm" /></TableCell>
+          <TableCell>
+            { phase.delay && (index!==props.phases.length-1) ?
+              ( 
                 <div>
                   <Input
-                    value={phase.delay}
+                    value={phase.delayUnit === 'days' ? Math.ceil(phase.delay/24) : phase.delay}
```

Comment:
given that we now also have minutes as an option I think we should just keep the `phase.delay`, otherwise it feels a bit strange that the input suddenly changes, but just when selecting 'days'

Reply:
Yep, good catch. Updated now!